bloom.go

```diff
@@ -0,0 +1,113 @@
+// Package bloom implements a Bloom filter data structure.
+package bloom
+
+import (
+	"crypto/sha256"
+	"encoding/binary"
+	"errors"
+	"fmt"
+	"math/bits"
+)
+
+// Filter is a Bloom filter, which represents a set of items and provides a probabilistic test for membership.
+// Filter satisfies the encoding.BinaryMarshaler and BinaryUnmarshaler interfaces.
+// The zero value represents an empty filter of size 0 that uses 0 hash values.
+type Filter struct {
+	f []byte
+	k int
+}
+
+// bit returns the filter's nth bit.
+func (f *Filter) bit(n int) int {
+	b, i := n/8, n%8
+	return int(f.f[b]) >> uint(i) & 1
+}
+
+// setBit sets the filter's nth bit to 1.
+func (f *Filter) setBit(n int) {
+	b, i := n/8, n%8
+	f.f[b] |= 1 << uint(i)
+}
+
+// New returns a Filter of size b bytes that uses k hash values.
+// It panics if b is not a power of 2 in the range [1, 8192] or k is not in the range [1, 16].
+func New(b, k int) *Filter {
+	if b <= 0 || b > 8192 {
+		panic("New: Filter size out of range")
+	}
+	if bits.OnesCount(uint(b)) != 1 {
+		panic("New: Filter size not a power of 2")
+	}
+	if k <= 0 || k > 16 {
+		panic("New: Number of hash values out of range")
```

Comment:
Nit: if you want to put a name at the front of the error string for context, it’s a little more conventional to to use the package name. Like “bloom: number of hash values out of range”. (Also https://github.com/golang/go/wiki/CodeReviewComments#error-strings error strings should not be capitalized.)